src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -454,34 +195,124 @@ private void PopulateKeywordArrays()
             {
                 KeywordFalse, KeywordTrue, KeywordParent, KeywordSelf
             };
+        }
+
+        public static TexlLexer GetLocalizedInstance(CultureInfo culture)
+        {
+            culture ??= CultureInfo.InvariantCulture;
+            return culture.NumberFormat.NumberDecimalSeparator == PunctuatorDecimalSeparatorInvariant ? InvariantLexer : CommaDecimalSeparatorLexer;
+        }
+
+        public static IList<string> GetKeywordDictionary()
+        {
+            IList<string> strList = new List<string>(_keywords.Count);
+
+            foreach (var keyword in _keywords.Keys)
+            {
+                strList.Add(keyword);
+            }
+
+            return strList;
+        }
+
+        private TexlLexer(string preferredDecimalSeparator)
+        {
+            // List and decimal separators.
+            // These are the default global settings. If there is a collision between the two,
+            // the list separator automatically becomes ;.
+            LocalizedPunctuatorDecimalSeparator = ChooseDecimalSeparator(preferredDecimalSeparator);
+            LocalizedPunctuatorListSeparator = ChooseListSeparatorPunctuator();
```

Comment:
>ChooseListSeparatorPunctuator [](http://example.com/codeflow?start=47&length=29)

There is an expected ordering of execution of this method - it needs to run after `LocalizedPunctuatorDecimalSeparator` is initialized - but it's not clear. You can pass the decimal separator as a parameter, and that would make the ordering explicit. Same for ChooseChainingPunctuator.